includes/course-theme/class-sensei-course-theme-lesson.php

```diff
@@ -142,4 +149,73 @@ private function maybe_add_prerequisite_notice() {
 			$notices->add_notice( 'locked_lesson', $text, __( 'You don\'t have access to this lesson', 'sensei-lms' ), [], 'lock' );
 		}
 	}
+
+	/**
+	 * Maybe add not enrolled notice.
+	 *
+	 * @return void
+	 */
+	private function maybe_add_not_enrolled_notice() {
+		$lesson_id                = \Sensei_Utils::get_current_lesson();
+		$course_id                = Sensei()->lesson->get_course_id( $lesson_id );
+		$notices                  = \Sensei_Context_Notices::instance( 'course_theme_locked_lesson' );
+		$registration_url         = sensei_user_registration_url() ?? wp_registration_url();
+		$sign_in_url              = add_query_arg( 'redirect_to', get_permalink(), $registration_url );
+		$take_course_redirect_url = add_query_arg( 'take_course_sign_in', '1', get_permalink( $course_id ) );
+		$take_course_url          = add_query_arg( 'redirect_to', $take_course_redirect_url, $registration_url );
+
+		if ( ! is_user_logged_in() ) {
+			$actions = [
+				[
+					'label' => __( 'Take course', 'sensei-lms' ),
+					'url'   => $take_course_url,
+					'style' => 'primary',
+				],
+				[
+					'label' => __( 'Sign in', 'sensei-lms' ),
+					'url'   => $sign_in_url,
+					'style' => 'secondary',
+				],
+			];
+
+			$notices->add_notice(
+				'locked_lesson',
+				__( 'Please register or sign in to access the course content.', 'sensei-lms' ),
+				__( 'You don\'t have access to this lesson', 'sensei-lms' ),
+				$actions,
+				'lock'
+			);
+
+			return;
+		}
+
+		if ( ! Sensei_Course::is_user_enrolled( $course_id ) ) {
+			if ( ! Sensei_Course::is_prerequisite_complete( $course_id ) ) {
+				$notices->add_notice(
+					'locked_lesson',
+					Sensei()->course::get_course_prerequisite_message( $course_id ),
+					__( 'You don\'t have access to this lesson', 'sensei-lms' ),
```

Comment:
Should this be "You don't have access to this course" instead?

Reply:
I think it should be "lesson" to keep consistent with the other messages because the other ones also talk about course access. cc @sruj09 to confirm. Here we're talking about:

<img width="940" alt="Screen Shot 2021-12-08 at 16 27 08" src="https://user-images.githubusercontent.com/876340/145272783-6159274e-9189-4823-b311-906819158e6b.png">

Also, good catch about the redirect issue. I'll explore a little because when using WP registration it worked for admin too. Probably it's something in our login.

Reply:
Confirming that it should be "lesson". :)
Can we tweak the pre-requisite message as well? "You must complete the prerequisites to view this lesson"

Reply:
Hey Srujana!

We can tweak this. Just before doing that, let me give you some context here to see what you think.

First, just to confirm that there's no confusion with the other prerequisite message, it's related to the Course prerequisite (the lesson is another message), okay? :)

The message we're using here is the same message being used on the Course page (with the blocks, and even before the blocks - in the "legacy version"). And this message is also associated with a filter, where the users can customize it.

I see 2 options we can go here:

1) Update the message that is being used on the Course page too.
2) Just ignore that, and use a custom message here (probably okay since it's a different scope - the lesson page).

Reply:
Got it. Yeah, the message sounds about right.

Reply:
I'm merging this PR, but we can continue the discussion and tweak it in a separate PR. ;)